test/floats.jl

```diff
@@ -366,6 +366,24 @@ end
 # discovered from JSON tests
 @test Parsers.tryparse(Float64, "0e+") === nothing
 
+@testset "groupmark" begin
+    @test Parsers.xparse(Float64, "100,000,000.99"; groupmark=',').val == 100_000_000.99
+    @test Parsers.xparse(Float64, "1,0,0,0,0,0,0,0,0.99"; groupmark=',').val == 100_000_000.99
+    @test Parsers.xparse(Float64, "100000000.99"; groupmark=',').val == 100_000_000.99
+
+    @test Parsers.xparse(Float32, "100,000,000.99"; groupmark=',').val ≈ 100_000_000.99
+    @test Parsers.xparse(Float32, "1,0,0,0,0,0,0,0,0.99"; groupmark=',').val ≈ 100_000_000.99
+    @test Parsers.xparse(Float32, "100000000.99"; groupmark=',').val ≈ 100_000_000.99
+
+    @test Parsers.xparse(Float64, "100,000,00099e-2"; groupmark=',').val == 100_000_000.99
+    @test Parsers.xparse(Float64, "1,0,0,0,0,0,0,0,099e-2"; groupmark=',').val == 100_000_000.99
+    @test Parsers.xparse(Float64, "10000000099e-2"; groupmark=',').val == 100_000_000.99
+
+    @test Parsers.xparse(Float32, "100,000,00099e-2"; groupmark=',').val ≈ 100_000_000.99
+    @test Parsers.xparse(Float32, "1,0,0,0,0,0,0,0,099e-2"; groupmark=',').val ≈ 100_000_000.99
+    @test Parsers.xparse(Float32, "10000000099e-2"; groupmark=',').val ≈ 100_000_000.99
```

Comment:
Tests look pretty good. Can we add a few more:
  * `@test_throws` when decimal and groupmark are the same?
  * Error/boundary cases; like `Parsers.xparse(Float64, "100,abc"; groupmark=',').val === 100.0`, and maybe that we didn't consume past the `,` as well (add a test for `tlen` on the parsers result)
  * Same error/boundary cases for ints
  * I think we also need to throw an error if `delim` and `groupmark` are the same; like if `delim=','` and `groupmark=','`, then `a,b,c,100,000` is ambiguous (columns of ["a", "b", "c", 100, 0] or ["a", "b", "c", 100_000])

Reply:
Great suggestions, thanks! 

For now I allow `groupmark == delim` iff `quoted` is true (but happy to change if you'd prefer). The reason is that a very common grouping mark is the comma, which is also a very common delimiter... and to disambiguate the two, one should use quoting.

This allows for a wider range of valid CSVs to be parsed, if people don't quote their locale-formatted numbers, this might produce weird results.

We seem to allow the following: 
```julia
julia> Parsers.xparse(Float64, "42,1"; decimal=',', delim=',')
Parsers.Result{Float64}(33, 4, 42.1)
```
which is also ambiguous and could happen in Czech locale. This is how Czech numbers are formatted:
```julia
Parsers.xparse(Float64, "1 042,1"; groupmark=' ', decimal=',')
```
I don't think `delim` is defined by locale, e.g. in Czech, conventionally `;` is the delimiter, but IIUC, this is not formalized (can't find it in [ICU explorer](https://icu4c-demos.unicode.org/icu-bin/locexp?d_=en&_=cs)) and a comma could be used as well (hopefully together with quoting).

Reply:
Ah yes, I should have clarified that if we're quoted, then we can allow delim == groupmark. Good call.